packages/aserto/src/aserto/client/directory/v2/__init__.py

```diff
@@ -19,23 +19,87 @@
 from aserto.client.directory.v2.helpers import ObjectIdentifier, RelationResponse
 
 
+def build_grpc_channel(address: str, default_address: str, ca_cert_path: str) -> Any:
+    if address == "" and default_address == "":
+        return None
+        
+    if address == "":
+        return grpc.secure_channel(
+        target=default_address, credentials=directory.channel_credentials(cert=ca_cert_path)
+    )
+        
+    return grpc.secure_channel(
+        target=address, credentials=directory.channel_credentials(cert=ca_cert_path)
+    )
+
 class Directory:
     def __init__(
         self,
         *,
-        address: str,
         api_key: str = "",
         tenant_id: str = "",
         ca_cert_path: str = "",
+        address: str = "",
+        reader_address: str = "",
+        writer_address: str = "",
+        importer_address: str = "",
+        exporter_address: str = "",
     ) -> None:
-        self._channel = grpc.secure_channel(
-            target=address, credentials=directory.channel_credentials(cert=ca_cert_path)
-        )
+        validation_error = directory.validate_addresses(address=address, reader_address=reader_address, writer_address=writer_address,
+                              importer_address=importer_address, exporter_address=exporter_address, model_address="")
+        if validation_error is not None:
+            raise validation_error
+
+        self._reader_channel = build_grpc_channel(reader_address, address, ca_cert_path)
```

Comment:
This approach means that even if all the services are listening on the same port, we would still create separate channels for each.
I'm pretty sure that in Go we create exactly one channel for each address.